src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] values;
+    private int size;
+
+    public ArrayList() {
+        values = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        increase();
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkAddIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            increase();
+            size++;
+            Object[] valuesTemp = new Object[size - index];
+            System.arraycopy(values, index, valuesTemp, 0, valuesTemp.length);
+            values[index] = value;
+            System.arraycopy(valuesTemp, 0, values, index + 1, valuesTemp.length - 1);
```

Comment:
you can copy elements in the same array
```suggestion
            System.arraycopy(values, index, values, index + 1, size - index);
            values[index] = value;
```